### PR TITLE
upgrade to nodejs 18.18.2

### LINKED ISSUE #83: Upgrade to Node 18+
We are 3 major versions of NodeJS behind.  xxwasm upgrade would be included, may need incremental upgrades first.  See #62 

Also:
> WARNING: node-v15.12.0 is in LTS Maintenance mode and nearing its end of life.

As part of this, address:
```shell
npm WARN deprecated @types/bson@4.2.0: This is a stub types definition. bson provides its own type definitions, so you do not need this installed.
npm WARN deprecated samsam@1.3.0: This package has been deprecated in favour of @sinonjs/samsam
```

### PR DESCRIPTION
**Pull request by shannonwells**
Problem
=======
Upgrade to nodejs 18+.  Closes #83 

Solution
========
Used last patch version of [nodejs 18](https://nodejs.org/en/about/previous-releases)

Change summary:
---------------
- [x] changed .tool-versions, 
- [x] changed package.json and 
- [x] ran `npm install` to update package-lock.json
- [x] updated all Git workflows
- [x] updated @types/node
- [x] updated README

Steps to Verify:
----------------
CI should pass

### CODE REVIEW

**Review by wilwade — CHANGES_REQUESTED** [review]
Found a few more places to update, but everything is working

- Ran locally
- Built
- Ran thrift

File: .tool-versions [hunk 1]
```diff
@@ -1 +1 @@
-nodejs 16.15.1
+nodejs 18.18.2
```
[the remark this answers is not in the document]
Comment (shannonwells) [reply .tool-versions:1]: I created another upgrade issue with these items in a checklist.

[the remark this answers is not in the document]
Comment (wilwade) [reply .tool-versions:1]: That sounds good. More of an Actions upgrade issue.

[the remark this answers is not in the document]
Comment (shannonwells) [reply .tool-versions:1]: looks like CI is ok with the actions versions for now.  maybe add it to the next upgrade issue?

**Review by wilwade — APPROVED** [review]
🚢

**Review by wilwade — APPROVED** [review]
Approving to make sure to capture all the changes

**Comment by 0Ams** [comment]
Why did you put a limit on the version?
I'm curious about the restrictions on the relief of 18.15.0 even though it operates normally

**Pull request merged**